test/astarte/client/appengine_test.exs

```diff
@@ -0,0 +1,46 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2021 SECO Mind
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Client.AppEngineTest do
+  use ExUnit.Case
+  doctest Astarte.Client.AppEngine
+
+  alias Astarte.Client.AppEngine
+
+  @valid_private_key X509.PrivateKey.new_ec(:secp256r1) |> X509.PrivateKey.to_pem()
+  @invalid_private_key "notaprivatekey"
+
+  describe "new" do
+    test "new/3 with jwt creates client" do
+      assert {:ok, %AppEngine{} = _client} =
+               AppEngine.new("https://base-url.com", "realm_name", jwt: "notarealjwt")
+    end
+
+    test "new/3 with valid private_key creates client" do
+      assert {:ok, %AppEngine{} = _client} =
+               AppEngine.new("https://base-url.com", "realm_name", private_key: @valid_private_key)
+    end
+
```

Comment:
When you implement the changes above, add a test to verify that explicitly passing `jwt` has precedence over the `private_key` (and that passing neither of them returns an error)